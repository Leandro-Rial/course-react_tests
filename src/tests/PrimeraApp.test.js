import React from "react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas PrimeraApp", () => {
  // test('debe mostrar el mensaje "Helouda"', () => {
  //     const saludo = "Helouda"
  //     const { getByText } = render( <PrimeraApp saludo={saludo} /> )
  //     expect( getByText(saludo) ).toBeInTheDocument()
  // })

  test("debe mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "Helouda";

    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el subtitulo enviado por props", () => {
    const saludo = "Helouda";
    const subTitulo = "+1 SUB";

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subTitulo} />
    );

    const textoParrafo = wrapper.find("p").text();

    expect(textoParrafo).toBe(subTitulo);
  });
});
