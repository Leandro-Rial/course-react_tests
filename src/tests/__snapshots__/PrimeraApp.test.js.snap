// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas PrimeraApp debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Helouda
  </h1>
  <p />
</Fragment>
`;
